server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java

```diff
@@ -53,6 +53,33 @@ protected static Set<String> getEnabledRemoteClusters(final Settings settings) {
         return RemoteConnectionStrategy.getRemoteClusters(settings);
     }
 
+    public static boolean isRemoteIndexName(String indexName) {
```

Comment:
Can you add javadoc here as to what input to send. The param should probably be called "indexExpression". I tend to think of "indexExpression" as possibly including a cluster alias, while to me indexName means just the index.

Also the javadoc should specify whether a "CSV list" is allowed, such as: `logs,mycluster:logs` or just a single indexExpression.

Reply:
As a non-review side question to the "CSV list" - when we are eventually checking for `::` separators in the near future, will we need to exhaustively check for singular `:` separators here in the case that a list of expressions is passed in? Otherwise, `index::data,cluster:index` would fail if we reach the first `::` and disqualify it as a remote index. Technically the example isn't a remote index expression, it is an expression that contains one, but the current logic would allow `index,cluster:index` through with a `true` result.

Reply:
Yes, this is supposed to be an individual index, but otherwise any other syntax is supposed to be supported.